lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -1140,6 +1140,9 @@ private function getFromClauses()
                 $tableSql = $from['table'];
                 $tableReference = $from['table'];
             } else {
+                if (array_key_exists($from['alias'], $knownAliases)) {
+                    throw QueryException::nonUniqueAlias($from['alias'], array_keys($knownAliases));
```

Comment:
You will have to revert this. This is a BC break for applications that rely on the current behaviour where defining a `FROM` clause with the same alias will overwrite the previous one. We cannot change this behaviour for now, even though it might have been a minor mistake in designing the query builder.